packages/frontend/src/components/SimplePieChart/PieChartContextMenu.tsx

```diff
@@ -16,40 +24,42 @@ export type PieChartContextMenuProps = {
         top: number;
     };
     value?: ResultValue;
+    groupDimensions?: string[];
+    rows?: ResultRow[];
 } & Pick<MenuProps, 'position' | 'opened' | 'onOpen' | 'onClose'>;
 
 const PieChartContextMenu: FC<PieChartContextMenuProps> = ({
     menuPosition,
     value,
+    groupDimensions,
+    rows,
     opened,
     onOpen,
     onClose,
 }) => {
+    const { projectUuid } = useParams<{ projectUuid: string }>();
     const { user } = useApp();
+    const { pieChartConfig } = useVisualizationContext();
     const { showToastSuccess } = useToaster();
     const clipboard = useClipboard({ timeout: 200 });
-    // const tracking = useTracking(true);
+    const tracking = useTracking(true);
     const metricQueryData = useMetricQueryDataContext(true);
 
-    const { projectUuid } = useParams<{ projectUuid: string }>();
-
-    if (!value || !metricQueryData) {
+    if (!value || !tracking || !metricQueryData) {
         return null;
     }
 
-    // const { openUnderlyingDataModal, openDrillDownModel } = metricQueryData;
-    // const { track } = tracking;
+    const { openUnderlyingDataModal /*, openDrillDownModel */ } =
+        metricQueryData;
+    const { track } = tracking;
 
-    const canViewUnderlyingData =
-        // TODO: implement this
-        false &&
-        user.data?.ability?.can(
-            'view',
-            subject('UnderlyingData', {
-                organizationUuid: user.data?.organizationUuid,
-                projectUuid: projectUuid,
-            }),
-        );
+    const canViewUnderlyingData = user.data?.ability?.can(
+        'view',
+        subject('UnderlyingData', {
+            organizationUuid: user.data?.organizationUuid,
```

Comment:
I think this should be more like 

`organizationUuid: metricQueryData.organiztionUuid` 

I don't know if that metricQueryData object has uuid, but I think you need to pass the orgUuid of the object in question and it gets compared to the users orgUuid. So passing the user's own uuid is a no-op.

I'm admittedly new to this thing though, so if I am not understanding something feel free to disagree 😄

Reply:
@magnew in this case we need user's organization uuid to check if user has an ability to use explore features. metricQueryData does not have organizationUuid

Reply:
I find the CASL pattern a bit confusing, but in a lot of places we have something like this:

```
const hasAdminAccess = user.ability.can(
        'manage',
        subject('Project', {
            organizationUuid: space.organizationUuid,
            projectUuid: space.projectUuid,
        }),
    );
```
Passing the uuid of the object in question. My understanding is that that uuid gets compared to the user uuid to see if they have the access to it. 

A fair amount of our frontend code does what you are doing (passed the users uuid to `subject`), but almost all of the backend code passes the uuid of the object in question. I assume we don't want totally different patterns for this on FE and BE, so one must be righter 😄.

I apologize if I am just reading something wrong.

Reply:
That's right it should ideally be something on the object that we're testing. For underlying data, it's consistent across the project. It is correct to pass in the projectUuid and organizationUuid of the _project_ and not of the user:

```typescript
{ organizationUuid, projectUuid } = useProject(...);
```

^ these are the correct values to pass into the check rather than using the user.

Reply:
updated, thanks both!